### PR TITLE
fix(sdk): fix ifpresent and concat placeholder compile

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Fixes compilation of ifpresent placeholder and uses the preferred ifpresent and concat placeholder representation in IR.

**Checklist:**
- [x] The title for your pull request (PR) should follow 
our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by connor-mccarthy** [comment]
/hold
_Merge after #7928._

File: sdk/python/kfp/components/pipeline_task.py [hunk 1]
```diff
@@ -232,13 +228,14 @@ def expand_command_part(arg) -> Union[str, List[str], None]:
                 if input_name in args or type_utils.is_task_final_status_type(
                         inputs_dict[input_name].type):
                     return arg.to_placeholder_string()
+
+                input_spec = inputs_dict[input_name]
+                if input_spec.default is None:
```
Comment (chensun) [inline sdk/python/kfp/components/pipeline_task.py:233]: I might miss some context here. Isn't default value always optional at compile time?

Comment (chensun) [reply sdk/python/kfp/components/pipeline_task.py:233]: Synced offline, this logic--requiring default--doesn't seem to be right, but it exists before this PR. We can go ahead merge this PR first and investigate this logic separately.

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**